### PR TITLE
[WIP] Feature errorprone compiler

### PR DESCRIPTION
**Pull request by olamy**
Enable errorprone compiler. ATM only warnings

### CODE REVIEW

**Comment by gregw** [comment]
For now, I think the errorprone check should be in a profile that is turned off by default.  Perhaps we run it on CI once per day/week?

**Comment by olamy** [comment]
sounds a good idea once per day.
But need I need to figure out how can we collect those errors in a more friendly way than reading console log...

**Comment by gregw** [comment]
See #2206

File: pom.xml [hunk 1]
```diff
@@ -406,9 +406,29 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.6.2</version>
           <configuration>
+            <showWarnings>true</showWarnings>
```
Comment (gregw) [inline pom.xml]: Move all this to a profile

File: jetty-util/src/main/java/org/eclipse/jetty/util/PathWatcher.java [hunk 2]
```diff
@@ -1029,35 +1030,34 @@ private void registerTree(Path dir, Config config, boolean notify) throws IOExce
         register(dir,config);
         
         final MultiException me = new MultiException();
-        Files.list(dir).forEach(p->
-        {
-            if (LOG.isDebugEnabled())
-                LOG.debug("registerTree? {}",p);
+        try (Stream<Path> stream = Files.list( dir)) {
```
Comment (gregw) [inline jetty-util/src/main/java/org/eclipse/jetty/util/PathWatcher.java]: wrong { formatting for jetty

**Pull request merged**